scheds/rust/scx_p2dq/src/bpf/main.bpf.c

```diff
@@ -302,37 +302,53 @@ static bool is_interactive(task_ctx *taskc)
 
 static bool can_migrate(task_ctx *taskc, struct llc_ctx *llcx)
 {
+	int idx;
+	u64 load, prev_load, non_intr_sum = 0;
+
 	if (nr_llcs <= 1 || (!dispatch_lb_interactive && taskc->interactive))
 		return false;
 
 	if (max_dsq_pick2 && taskc->dsq_index != nr_dsqs_per_llc - 1)
 		return false;
 
-	u64 mig_nr_queued = scx_bpf_dsq_nr_queued(llcx->mig_dsq);
-
-	// The first queued migration tasks should be the least interactive
-	if (mig_nr_queued == 0 && taskc->dsq_index != nr_dsqs_per_llc - 1)
-		return false;
-
-	// If the number of queued tasks is less than half of the number of
-	// CPUs then only try to migrate least interactive tasks.
-	if (taskc->dsq_index != nr_dsqs_per_llc - 1 &&
-	    mig_nr_queued > llcx->nr_cpus / (llcx->nr_cpus > 1 ? 2 : 1))
+	if ((taskc->all_cpus && taskc->llc_runs < min_llc_runs_pick2) ||
+	     !taskc->all_cpus)
 		return false;
 
+	u64 mig_nr_queued = scx_bpf_dsq_nr_queued(llcx->mig_dsq);
 	u64 llc_nr_queued = scx_bpf_dsq_nr_queued(llcx->dsq);
-	if (mig_nr_queued >= llc_nr_queued || mig_nr_queued >= llcx->nr_cpus)
+	if (mig_nr_queued >= llc_nr_queued)
 		return false;
 
 	if (interactive_dsq) {
 		u64 intr_nr_queued = scx_bpf_dsq_nr_queued(llcx->intr_dsq);
-		if (mig_nr_queued > intr_nr_queued)
+		if (mig_nr_queued >= intr_nr_queued)
 			return false;
 	}
 
-	return (nr_llcs > 1 &&
-		taskc->all_cpus &&
-		taskc->llc_runs > min_llc_runs_pick2);
+	// The least interactive can always migrate
+	if (taskc->dsq_index == MAX_DSQS_PER_LLC - 1)
+		return true;
```

Comment:
Does this mean "is it the migration DSQ?" now?

Reply:
I guess so, there's still the check that the task has run sufficiently in the LLC prior to this to prevent the least interactive tasks from ping-ponging.